server/src/postgres/simple_resource_linked.rs

```diff
@@ -0,0 +1,539 @@
+use super::{util::*, ArrErr, PsqlData, PsqlField, PsqlFieldSend};
+use crate::grpc::server::ValidationResult;
+use crate::grpc::{GrpcDataObjectType, GrpcField};
+use crate::resources::base::simple_resource::*;
+
+use chrono::{DateTime, Utc};
+use deadpool_postgres::Transaction;
+use geo_types::{LineString, Point, Polygon};
+use std::collections::HashMap;
+use std::vec;
+use tokio_postgres::types::Type as PsqlFieldType;
+use tokio_postgres::Row;
+use uuid::Uuid;
+
+/// Generic PostgreSQL trait to provide wrappers for common `Resource` functions
+#[tonic::async_trait]
+pub trait PsqlType
+where
+    Self: Resource + Clone + Sized,
+{
+    /// Generic get for id function to get rows for the provided key fields
+    /// Since this is a linked resource, the id is expected to be given as a [Vec\<FieldValuePair\>]
+    /// to specify the id_column / value pairs to match
+    /// The provided ids should be a combined primary key, so just one result should
+    /// returned.
+    async fn get_for_ids(ids: HashMap<String, Uuid>) -> Result<Row, ArrErr> {
+        psql_debug!("(get_for_ids) start: [{:?}]", ids);
+        super::queries::get_for_ids::<Self>(&ids).await
+    }
+
+    /// Generic delete for ids function to delete rows for the provided key fields
+    /// Since this is a linked resource, the id is expected to be given as a [Vec\<FieldValuePair\>]
+    /// to specify the id_column / value pairs to match
+    /// An optional transaction handle can be provided, which will be used if present.
+    /// This function will not commit, so the caller is responsible for committing the transaction when done.
+    async fn delete_for_ids(
+        ids: HashMap<String, Uuid>,
+        transaction: Option<&Transaction>,
+    ) -> Result<(), ArrErr> {
+        psql_debug!("(delete_for_ids) start: [{:?}]", ids);
+        let definition = Self::get_definition();
+
+        let mut params: Vec<Box<PsqlFieldSend>> = vec![];
+        let mut query = format!(r#"DELETE FROM "{}""#, definition.get_psql_table());
+        let mut search_operator = "WHERE";
+        let mut next_param_index: i32 = 1;
+
+        for (field, value) in ids.clone() {
+            if Self::has_id_col(&field) {
+                query.push_str(&format!(
+                    r#" {} "{}" = ${}"#,
+                    search_operator, field, next_param_index
+                ));
+                params.push(Box::new(value));
+                search_operator = "AND";
+                next_param_index += 1;
+            }
+        }
+
+        psql_debug!("{}", &query);
+        psql_debug!("{:?}", &params);
+        psql_info!(
+            "Deleting rows for table [{}]. uuids: {:?}",
+            definition.psql_table,
+            ids
+        );
+
+        let mut ref_params: Vec<&PsqlField> = vec![];
+        for field in params.iter() {
+            ref_params.push(field.as_ref());
+        }
+
+        // TODO(R3): Move this to 2 separate functions which can be used in other places as well
```

Comment:
Got some TODO(R3) throughout:

<details>

Test  | Location                                            | Statement                                         
----- | --------------------------------------------------- | --------------------------------------------------
  X   | server/src/postgres/linked_resource.rs (71)         | // TODO(R3): Move this to 2 separate functions ...
  X   | server/src/postgres/linked_resource.rs (164)        | /// TODO: use `#![feature(async_fn_in_trait)]` ...
  X   | server/src/postgres/linked_resource.rs (175)        | //TODO(R3): implement shared memcache here to g...
  X   | server/src/postgres/linked_resource.rs (256)        | //TODO(R3): flush shared memcache for this reso...
  X   | server/src/postgres/simple_resource_linked.rs (76)  | // TODO(R3): Move this to 2 separate functions ...
  X   | server/src/postgres/simple_resource_linked.rs (152) | /// TODO: use `#![feature(async_fn_in_trait)]` ...
  X   | server/src/postgres/simple_resource_linked.rs (166) | //TODO(R3): implement shared memcache here to g...
  X   | server/src/postgres/simple_resource_linked.rs (219) | //TODO(R3): flush shared memcache for this reso...
  X   | server/src/postgres/simple_resource_linked.rs (294) | //TODO(R3): flush shared memcache for this reso...
  X   | server/src/postgres/simple_resource_linked.rs (338) | //TODO(R3): flush shared memcache for this reso...
  X   | server/src/postgres/mod.rs (223)                    | //TODO(R3): provide metrics, eg: self.metrics.p...
  X   | server/src/postgres/simple_resource.rs (84)         | /// TODO: use `#![feature(async_fn_in_trait)]` ...
  X   | server/src/postgres/simple_resource.rs (98)         | //TODO(R3): implement shared memcache here to g...
  X   | server/src/postgres/simple_resource.rs (151)        | //TODO(R3): flush shared memcache for this reso...
  X   | server/src/postgres/simple_resource.rs (226)        | //TODO(R3): flush shared memcache for this reso...
  X   | server/src/postgres/simple_resource.rs (270)        | //TODO(R3): flush shared memcache for this reso...
  X   | server/src/resources/scanner/mod.rs (73)            | // TODO(R3) After groups are implemented, add o...
  X   | client-grpc/examples/grpc.rs (184)                  | // Get acquire and deliver fields - TODO 

</details>